waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -43,38 +50,20 @@ proc createRLNInstance*(d: int = MERKLE_TREE_DEPTH): RLNResult
   ## An RLN instance supports both zkSNARKs logics and Merkle tree data structure and operations
   ## d indicates the depth of Merkle tree
   var
-    rlnInstance: RLN[Bn256]
+    rlnInstance: ptr RLN
     merkleDepth: csize_t = uint(d)
-    ## parameters.key contains the prover and verifier keys
-    ## to generate this file, clone this repo https://github.com/kilic/rln
-    ## and run the following command in the root directory of the cloned project
-    ## cargo run --example export_test_keys
-    ## the file is generated separately and copied here
-    ## parameters are function of tree depth and poseidon hasher
-    ## to generate parameters for a different tree depth, change the tree size in the following line of rln library
-    ## https://github.com/kilic/rln/blob/3bbec368a4adc68cd5f9bfae80b17e1bbb4ef373/examples/export_test_keys/main.rs#L4
-    ## and then proceed as explained above
-    parameters = readFile("waku/v2/protocol/waku_rln_relay/parameters.key")
-    pbytes = parameters.toBytes()
-    len: csize_t = uint(pbytes.len)
-    parametersBuffer = Buffer(`ptr`: addr(pbytes[0]), len: len)
-
-  # check the parameters.key is not empty
-  if (pbytes.len == 0):
-    debug "error in parameters.key"
-    return err("error in parameters.key")
+    resourcesPathBuffer = RLN_RESOURCE_FOLDER.toOpenArrayByte(0, RLN_RESOURCE_FOLDER.high).toBuffer()
```

Comment:
I'd do error handling on this, assume rln resource folder can't be found / is null

Reply:
The error handling is done in zerokit RLN (cf. [1](https://github.com/vacp2p/zerokit/blob/9cc333811b0a44638fe5ea5ac26a11f91a15aa68/rln/src/circuit.rs#L40), [2](https://github.com/vacp2p/zerokit/blob/9cc333811b0a44638fe5ea5ac26a11f91a15aa68/rln/src/circuit.rs#L58-L60), [3](https://github.com/gakonst/ark-circom/blob/a93c8b03d4376e197ceb36aa410a8903c283dab7/src/circom/builder.rs#L28-L29)) but we might add an extra error handling layer in nwaku as well if better.